### DISCUSSION TITLE
It'd be nice if could pass attrs from parent to child at compose time

### DISCUSSION THREAD

**Discussion by neilmcguigan** [jpsca/jinjax#14] [Q&A] [answered]
Say I want a generic container component that can dynamically set args on declared child components

like this:

```
<Parent>
  <Child {someAttrsFromParent} />
  <Child />
</Parent>
```

You can do this w Jinja macros (awkwardly):

```
{% call(kwargs) parent() %}
  {{ child(foo=1, **kwargs) }}
  {{ child(foo=2, **kwargs) }}
{% endcall %}
```

where parent looks like this:

```
{% macro parent() %}
...
{{  caller( dict(bar=2) ) }}
...
{% endmacro %}
```

**Comment by jpsca** [comment, marked as answer]
You can! But it's barely mentioned in the docs :( (https://jinjax.scaletti.dev/guide/extra/#attrs-methods).

```html+jinja
<Parent>
  <Child __attrs={attrs} />
  <Child />
</Parent>
```
1. `attrs` are the parent attrs
2. `__attrs` begins with **two** underscores

If you only want to pass _some_ arguments (and you know which ones), you can do it like any other value (`attrs` is similar to a dict)

```html+jinja
<Parent>
  <Child title={attrs.get('title') or 'Hello'} />
  <Child />
</Parent>
```

**Reply by CiberNin** [reply]
Is something like a React context provider possible?
```jinja GritIndexPage.jinja
{# def model#}
<Datagrid mode={model}>
  <Field name="id">
  <Field name="grid">
</Datagrid>
```
Where model is an sqlmodel that each Field wants to introspect to find out it's datatype so it can render differently. And the Datagrid component somehow internally handles passing the model to each of it's children.
